realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/UserTests.kt

```diff
@@ -183,34 +179,34 @@ class UserTests {
         assertTrue(anonUser === linkedUser)
         assertEquals(2, linkedUser.identities.size)
         assertEquals(Credentials.IdentityProvider.EMAIL_PASSWORD, linkedUser.identities[1].provider)
-        admin.setAutomaticConfirmation(enabled = true)
 
         val otherEmail = TestHelper.getRandomEmail()
         val otherPassword = "123456"
         app.emailPasswordAuth.registerUser(otherEmail, otherPassword)
-        linkedUser = anonUser.linkCredentials(Credentials.emailPassword(email, password))
-        assertTrue(anonUser === linkedUser)
-        assertEquals(3, linkedUser.identities.size)
-        assertEquals(Credentials.IdentityProvider.EMAIL_PASSWORD, linkedUser.identities[2].provider)
+
+        val credentials = Credentials.emailPassword(otherEmail, otherPassword)
+
+        // Validate that we cannot link a second set of credentials
+        assertFails {
+            linkedUser = anonUser.linkCredentials(credentials)
+        }
+
         admin.setAutomaticConfirmation(enabled = true)
     }
 
-    @Ignore("FIXME: Wait for linkUser support in ObjectStore")
     @Test
     fun linkUser_existingCredentialsThrows() {
         val email = TestHelper.getRandomEmail()
         val password = "123456"
         val emailUser: User = app.registerUserAndLogin(email, password)
-        val anonymousUser: User = app.login(Credentials.anonymous())
         try {
-            anonymousUser.linkCredentials(Credentials.emailPassword(email, password))
+            anonUser.linkCredentials(Credentials.emailPassword(email, password))
             fail()
         } catch (ex: AppException) {
-            assertEquals(ErrorCode.BAD_REQUEST, ex.errorCode)
+            assertEquals(ErrorCode.INVALID_SESSION, ex.errorCode)
```

Comment:
While testing the error the app returns is an INVALID_SESSION instead of a BAD_REQUEST. 

Changed the test to match the returning error.